app/controllers/api_new/users_controller.rb

```diff
@@ -12,4 +12,9 @@ class ApiNew::UsersController < ApiNew::ApiController
 	def current
 		@user = current_user
 	end
+
+	# get /api_new/users/current_nonprofit/object_events
+	def current_nonprofit_object_events
+		redirect_to "/api_new/nonprofits/#{current_user.roles.where(host_type: 'Nonprofit').first&.host&.houid}/object_events"
```

Comment:
You need to pass the parameters along as part of the redirect. The Rails route helper method makes that easier.